sdk/communication/Azure.Communication.CallingServer/tests/CallRecording/CallRecordingAutomatedLiveTests.cs

```diff
@@ -0,0 +1,195 @@
+                var uniqueId = await ServiceBusWithNewCall(user, target);
+
+                // create call and assert response
+                CreateCallResult response = await client.CreateCallAsync(new CallSource(user), new CommunicationIdentifier[] { target }, new Uri(TestEnvironment.DispatcherCallback + $"?q={uniqueId}")).ConfigureAwait(false);
+                string callConnectionId = response.CallConnectionProperties.CallConnectionId;
+                Assert.IsNotEmpty(response.CallConnectionProperties.CallConnectionId);
+
+                // wait for incomingcall context
+                string? incomingCallContext = await WaitForIncomingCallContext(uniqueId, TimeSpan.FromSeconds(20));
+                Assert.IsNotNull(incomingCallContext);
+
+                // answer the call
+                var answerResponse = await client.AnswerCallAsync(incomingCallContext, new Uri(TestEnvironment.DispatcherCallback));
+                Assert.AreEqual(answerResponse.GetRawResponse().Status, StatusCodes.Status200OK);
+
+                // wait for callConnected
+                var connectedEvent = await WaitForEvent<CallConnected>(callConnectionId, TimeSpan.FromSeconds(20));
+                Assert.IsNotNull(connectedEvent);
+                Assert.IsTrue(connectedEvent is CallConnected);
+                Assert.IsTrue(((CallConnected)connectedEvent!).CallConnectionId == callConnectionId);
+
+                // test get properties
+                Response<CallConnectionProperties> properties = await response.CallConnection.GetCallConnectionPropertiesAsync().ConfigureAwait(false);
+                Assert.AreEqual(CallConnectionState.Connected, properties.Value.CallConnectionState);
+
+                // try start recording unmixed audio - no channel affinity
+                var startRecordingResponse = await client.GetCallRecording().StartRecordingAsync(
+                    new StartRecordingOptions(new ServerCallLocator(properties.Value.ServerCallId))
+                    {
+                        RecordingChannel = RecordingChannel.Unmixed,
+                        RecordingContent = RecordingContent.Audio,
+                        RecordingFormat = RecordingFormat.Wav,
+                        // RecordingStateCallbackEndpoint = "DISPATCHER CALLBACK URI?" TODO: Add Dispatcher's callback URI here, since this is OutOfCall API
+                    });
+                Assert.AreEqual(StatusCodes.Status200OK, startRecordingResponse.GetRawResponse().Status);
+                Assert.NotNull(startRecordingResponse.Value.RecordingId);
+
+                // try stop recording
+                var stopRecordingResponse = await client.GetCallRecording().StopRecordingAsync(startRecordingResponse.Value.RecordingId);
+                Assert.AreEqual(StatusCodes.Status204NoContent, stopRecordingResponse.Status);
+
+                // wait for CallRecordingStateChanged event TODO: Dispatcher should receive this callback event
```

Comment:
Not sure if Dispatcher needs to be updated to receive the CallRecordingStateChanged event